### PR TITLE
Change Shutdown() and ForceFlush() to return bool for traces

### PR DESCRIPTION
**Pull request by ThomsonTan**
This resolves https://github.com/open-telemetry/opentelemetry-cpp/issues/400. The default timeout is also set to `std::chrono::microseconds::max()` as the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#shutdown-1) requires `Shutdown() Shutdown should not block indefinitely`.

### CODE REVIEW

File: exporters/otlp/include/opentelemetry/exporters/otlp/otlp_exporter.h
Comment (jsuereth) [inline exporters/otlp/include/opentelemetry/exporters/otlp/otlp_exporter.h:56]: I think this is a syntax error, right?

```
return true;}
```

File: exporters/otlp/include/opentelemetry/exporters/otlp/otlp_exporter.h
Comment (jsuereth) [reply exporters/otlp/include/opentelemetry/exporters/otlp/otlp_exporter.h:55]: +1 We shouldn't specify defaults in overridden methods.  AFAIK it should pull the default from the base type.

**Review by jsuereth — CHANGES_REQUESTED** [review]
+1 to returning `bool`, looks like one syntax error is breaking tests.

File: exporters/otlp/include/opentelemetry/exporters/otlp/otlp_exporter.h
Comment (ThomsonTan) [reply exporters/otlp/include/opentelemetry/exporters/otlp/otlp_exporter.h]: Seems overrider doesn't acquire the defaults from base class.

>8.3.6.10:
A virtual function call (10.3) uses the default arguments in the declaration of the virtual function determined by the static type of the pointer or reference denoting the object. An overriding function in a derived class does not acquire default arguments from the function it overrides. Example:

File: exporters/otlp/include/opentelemetry/exporters/otlp/otlp_exporter.h
Comment (ThomsonTan) [reply exporters/otlp/include/opentelemetry/exporters/otlp/otlp_exporter.h]: Good catch, missed this as my local build did enable OTLP.

**Review by lalitb — APPROVED** [review]
LGTM

**Comment by alolita** [comment]
Triaged Dec 7 2020: Need another reviewer to complete review. @jsuereth can you please take a look?

**Review by pyohannes — APPROVED** [review]

**Comment by pyohannes** [comment]
Approved, but before merging we should have a look why the Bazel asan test fails (maybe it's just a temporary issue).

**Comment by lalitb** [comment]
> Approved, but before merging we should have a look why the Bazel asan test fails (maybe it's just a temporary issue).

@ThomsonTan  Do you think if error in bazel asan test is related to this PR. Seems its failing consistently: 
`/usr/include/c++/7/chrono:176:38: runtime error: signed integer overflow: 9223372036854775807 * 1000 cannot be represented in type 'long int'`

**Comment by ThomsonTan** [comment]
> > Approved, but before merging we should have a look why the Bazel asan test fails (maybe it's just a temporary issue).
> 
> @ThomsonTan Do you think if error in bazel asan test is related to this PR. Seems its failing consistently:
> `/usr/include/c++/7/chrono:176:38: runtime error: signed integer overflow: 9223372036854775807 * 1000 cannot be represented in type 'long int'`

Looking into ASAN the failure.

**Comment by ThomsonTan** [comment]
The ASAN failure was caused by assigned `std::chrono::milliseconds` to `std::chrono::microseconds`. Fixed in the latest commit.

**Comment by lalitb** [comment]
> The ASAN failure was caused by assigned `std::chrono::milliseconds` to `std::chrono::microseconds`. Fixed in the latest commit.

Thanks @ThomsonTan . @jsuereth  its blocked for your review now : )

**Pull request merged**
